### ISSUE TITLE
Authentication Not Sticking

### ISSUE THREAD

**Issue by danda1man** [thinktecture/Thinktecture.IdentityServer.v3#780]
I updated to that latest release over the weekend and it's causing the authentication to not stick.  I was using the hybrid flow, but I thought something may have changed so I stripped it down to the example MVC project (Implicit) to see if I could get it working again, no luck.  Authorize on my controller kicks me out to the login page, I authenticate, and then there is an infinite loop on the form_post.  I set breakpoints in the SecurityTokenValidated method, and everything is there and the principal is authenticated, but the redirect gets back an unauthorized and it keeps cycling.  I turned up logging to Trace and pasted a cycle below, everything looks ok there for me too:

![image](https://cloud.githubusercontent.com/assets/852995/5803329/1b6bc06a-9fc3-11e4-8b7e-4c7d3e8a39d3.png)

**Comment by brockallen** [comment]
Disable `UseTokenLifetime` on the OIDC middleware and see if things improve.

**Comment by danda1man** [comment]
UseTokenLifetime is already set to false:

![image](https://cloud.githubusercontent.com/assets/852995/5804370/fe6c4fae-9fca-11e4-9abd-599038be8aa0.png)

**Comment by brockallen** [comment]
So then look into the other things I suggested. You'll need to debug the HTTP traffic to understand why it's not working.

**Comment by danda1man** [comment]
Suggested where?

**Comment by brockallen** [comment]
From here: https://github.com/thinktecture/Thinktecture.IdentityServer.v3/issues/780#issuecomment-70526927

**Comment by danda1man** [comment]
Oh I see, my page didn't have the full comment until I refreshed.  I never get logged in so I am looking into the VS cert, I did catch this in elmah though, suppose that could just be the cookie not sticking though:

IDX10311: RequireNonce is 'true' (default) but validationContext.Nonce is null. A nonce cannot be validated. If you don't need to check the nonce, set OpenIdConnectProtocolValidator.RequireNonce to 'false'.
